controllers/servicedefaults_controller.go

```diff
@@ -149,26 +170,50 @@ func (r *ServiceDefaultsReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func syncFailed(reason, message string) consulv1alpha1.Conditions {
-	return consulv1alpha1.Conditions{
+// consulNamespace returns the namespace that a service should be
+// registered in based on the namespace options. It returns an
+// empty string if namespaces aren't enabled.
+func (r *ServiceDefaultsReconciler) consulNamespace(ns string) string {
```

Comment:
This function is also copied across all files however I could see the destination namespace logic changing depending on location so that's why I didn't de-duplicate this and throw it in the new `namespaces` package.

Reply:
Do you have an example in mind of how you think this could change? Do you think it will hard to de-duplicate when we need to?

Reply:
The answer to your question leads me to refactor this into `namespaces` :D brb

Reply:
Updated